app/src/main/java/com/github/se/polyfit/data/api/ProductInfoResponseAPI.kt

```diff
@@ -0,0 +1,42 @@
+package com.github.se.polyfit.data.api
+
+import com.github.se.polyfit.model.nutritionalInformation.MeasurementUnit
+import com.github.se.polyfit.model.nutritionalInformation.Nutrient
+import com.github.se.polyfit.model.nutritionalInformation.NutritionalInformation
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+
+class ProductResponseAPI(val productResponse: ProductResponse?, val status: APIResponse) {
+  companion object {
+    val moshi: Moshi by lazy { Moshi.Builder().add(KotlinJsonAdapterFactory()).build() }
+
+    fun fromJson(jsonString: String): ProductResponse {
+      val adapter = moshi.adapter(ProductResponse::class.java).lenient()
```

Comment:
Nitpicking here, but it could be a good idea to move this outside the function so that it is not created and initialized every time the function is called.